### ISSUE TITLE
Single unboxed `tabSet` values in `panes` config leads to startup errors

### ISSUE THREAD

**Issue by pat-s** [rstudio/rstudio#6641]
### System details

    RStudio Edition : Desktop
    OS Version      : 1.4.170
    R Version       : 3.6.3

### Steps to reproduce the problem

Use the following setting in `rstudio-prefs.json`:

```json
{
  "panes": {
    "quadrants": ["Source", "TabSet1", "Console", "TabSet2"],
    "tabSet1": "Presentation",
    "tabSet2": ["Environment", "History", "Files", "Plots", "Packages", "Help", "Build", "VCS", "Viewer"],
    "hiddenTabSet": ["Connections", "Tutorial"],
    "console_left_on_top": false,
    "console_right_on_top": true
  }
}
```

### Describe the problem in detail

`rstudio-prefs.json` settings need to unboxed by default otherwise they are ignored.

Using functions to write JSON files with automatic unboxing such as `jsonlite::write_json(auto_unbox = TRUE)`, `tabSet*` values with single entries in the panes category also get unboxed if their length is <= 1 (e.g. tabSet1 in the example above).
This applies if a tabSet has only one entry which happens if a certain tabSet is hidden (which is supported for RStudio > 1.4.162).

With single unboxed values of tabSet settings, RStudio errors during startup.

### Describe the behavior you expected

RStudio should be able to deal with unboxed scalars in the tabSet configuration like it does for other settings and startup normally.

---

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Is this configuration that you wrote yourself, or was it generated by the IDE? If this is something you're editing by hand, I'd recommend just ensuring it is a single-element array.

**Comment by pat-s** [comment]
> Is this configuration that you wrote yourself, or was it generated by the IDE?

I wrote it myself. To be more specific its scripted so I can apply the config across different OS. I am using `jsonlite::write_json()` and because I need to unbox the values during writing via `auto_unbox = TRUE`, I am running into this issue.

Would be really great if unboxed scalars would not error in the in the `panes` category.
Let me know if you plan to fix it - if not, I'll probably take the time to adjust for this manually. However, I would rather not to if I know that this is going to be fixed upstream :)

**Comment by kevinushey** [comment]
It's unlikely that we'll change the behavior here. I'd recommend using `I()` to protect things that need to be arrays from unboxing, e.g.

```
> jsonlite::toJSON(list(a = 1, b = I(2)), auto_unbox = TRUE)
{"a":1,"b":[2]} 
```

**Comment by pat-s** [comment]
Ah! Very helpful. Thanks.

After 40 mins I realized the issue I was stumbling over: When reading the JSON file, I used `jsonlite::fromJSON()` which simplifies by default and always unboxes scalars automatically.

Switching to `jsonlite::read_json()` or using `jsonlite::fromJSON(simplifyVector = FALSE)` prevents the unboxing of scalars.

So in the end if there is a valid config and one uses the correct JSON parser functions, everything is fine. So this issue was on my side - sorry!
